lib/nexmo_markdown_renderer/models/code_language.rb

```diff
@@ -64,6 +64,21 @@ def self.linkable
       def self.route_constraint
         { code_language: Regexp.new(linkable.map(&:key).compact.join('|')) }
       end
+
+      def self.filter_languages_for_dropdown(languages, item_list)
+        languages_mapped = []
+
```

Comment:
@benhayehudi i don't think you really need this method. if i understand correctly, you want to return the languages that are associated to the `item_list`, so instead you can do:

```ruby
item_list.map(&:languages).flatten.uniq.map(&:downcase)
```

and given that you want to have the intersection, you can use the result of that and do `result & languages`

Also, I think this should live in a presenter in NDP, it's mostly for presentation purposes and only use there. WFYT?